### PR TITLE
Add extra options for slack webhook operator and slack hook

### PR DESCRIPTION
**Pull request by benbenbang**
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target [Github ISSUE](https://github.com/apache/airflow/issues/9407) in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by benbenbang** [comment]
Do I miss something here?
```
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.```

**Review by mik-laj — APPROVED** [review]

File: airflow/contrib/hooks/slack_webhook_hook.py [hunk 1]
```diff
@@ -86,6 +89,7 @@ def __init__(self,
         self.icon_url = icon_url
         self.link_names = link_names
         self.proxy = proxy
+        self.extra_options = extra_options or dict()
```
Comment (kaxil) [inline airflow/contrib/hooks/slack_webhook_hook.py]:
```suggestion
        self.extra_options = extra_options or {}
```

**Review by kaxil — APPROVED** [review]
Minor comment but LGTM otherwise

**Comment by benbenbang** [comment]
Hi @kaxil, @mik-laj 👋 
CI Build / Core:Pg9.6,Py3.7 failed without any logs, is it bcs time out? Thanks

**Comment by kaxil** [comment]
I have restarted CI, let's see if it passes this time

**Comment by benbenbang** [comment]
hmmm, it failed randomly... Do you have any thoughts on this @kaxil?

**Review by dimberman — APPROVED** [review]

**Pull request merged**